homeassistant/components/knx/knx_entity.py

```diff
@@ -2,30 +2,52 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any
 
 from xknx.devices import Device as XknxDevice
 
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity import Entity
-
-from .const import DOMAIN
+from homeassistant.helpers.entity_platform import EntityPlatform
 
 if TYPE_CHECKING:
     from . import KNXModule
 
-SIGNAL_ENTITY_REMOVE = f"{DOMAIN}_entity_remove_signal.{{}}"
+
+class KnxUiEntityPlatformController:
+    """Class to manage dynamic adding and reloading of UI entities."""
+
+    def __init__(
+        self,
+        knx_module: KNXModule,
+        entity_platform: EntityPlatform,
+        entity_class: type[KnxUiEntity],
+    ) -> None:
+        """Initialize the UI platform."""
+        self._knx_module = knx_module
+        self._entity_platform = entity_platform
+        self._entity_class = entity_class
+
+    async def add_entity(self, unique_id: str, config: dict[str, Any]) -> None:
+        """Add a new UI entity."""
+        await self._entity_platform.async_add_entities(
+            [self._entity_class(self._knx_module, unique_id, config)]
+        )
+
+    async def reload_entity(
```

Comment:
I think it's better to replace this with `remove_entity` and let the storage first call `remove_entity` and then `add_entity`.

Reply:
I didn't do that, so it can't be mistakenly used to delete entities - since this should be done through the entity registry which takes care of removing the entity itself.

Maybe `reload_with_new_config()` or something would be a better name 🤔

Reply:
If you want to keep this method you should pass an entity registry entry instead of both entity_id and unique_id. It's weird to pass two different ids for the same thing.